testing/acceptance_test.py

```diff
@@ -822,8 +822,8 @@ def test_cmdline_python_package_symlink(self, testdir, monkeypatch):
         if hasattr(py.path.local, "mksymlinkto"):
             result.stdout.fnmatch_lines(
                 [
-                    "lib/foo/bar/test_bar.py::test_bar PASSED*",
-                    "lib/foo/bar/test_bar.py::test_other PASSED*",
+                    "local/lib/foo/bar/test_bar.py::test_bar PASSED*",
+                    "local/lib/foo/bar/test_bar.py::test_other PASSED*",
```

Comment:
Note that the behavior here also changes (due to https://github.com/pytest-dev/pytest/pull/4159 however already, which should have changed it to using `-vv` instead probably).

Also reminded me of https://github.com/pytest-dev/pytest/pull/6701/files#diff-1f1fc4eef6bf4c4da558e0dce9e74e9b - which is a bit related maybe at least?

Reply:
Answering both comments here...

I didn't comment on this because it's actually the same thing (i.e.: it reports things based on the view of the working dir, which includes the symlinked version and not the resolved version).

I tried with https://github.com/pytest-dev/pytest/pull/6701 and tests fail -- the reason is still the same: with this change `pytest` views things are they are from the current working directory (i.e.: symlinked), not from the resolved version...

So, basically it boils down to the way `pytest` should see things... it should either ignore links and always use the resolved version (and symlinks are a way to `cd` into a module) or symlinks are respected and that becomes the way that `pytest` views things.

If symlinks are to be respected, I don't think pytest should resolve `conftest` if it's not seen from the working directory (it makes things more complex than they should -- and I'd find that surprising if it works)... as you mentioned, if that's needed the user should also symlink `conftest`.

-- as a note, the main issue with resolving is that the paths reported aren't the paths that I'm seeing on IDEs/editors, so, clicking to get to the source file gets a different version of the file (i.e.: I've opened `/my/linked/folder/my_file.py` and paths are seen/reported as `/my/real/folder/my_file.py`, which opens a different view of the file and confuses the IDEs/editors opening a second version of the file I already have open) -- the fact that the full path is also reported on the console is also a bit annoying but not as much as having IDEs/editors confused (I checked with PyDev and VSCode Python, and I'd be surprised if any IDE would be able to find it... the solution is of course never using symlinks/subst drives on IDEs and always use the real folders, but that's pretty annoying if you actually use those to structure your code/projects).

Reply:
@blueyed @nicoddemus any thoughts here?

Reply:
I honestly don't know the code well enough to have thoughts on the above, except to agree that, on Windows, most apps don't "resolve" link-like things, such as subst drives.

I'm still pegged to pytest==5.0.1 to avoid this bug.